Add template engine example (#151)
encoredev/examples: ts/template-engine/README.md

# Template Engine with Encore.ts

> This example utilizes a Raw endpoint to handle serve views through a template engine. This example uses the [EJS](https://ejs.co/) for rendering views but the same concept could be used for a different template engine.

Reviewer: ```suggestion
This example utilizes a Raw endpoint to serve views through a template engine. The example implementation uses [EJS](https://ejs.co/) for rendering views, but the same approach is also applicable for other template engines.
```

## Developing locally

### Prerequisite: Installing Encore

If this is the first time you're using Encore, you first need to install the CLI that runs the local development
environment. Use the appropriate command for your system:

- **macOS:** `brew install encoredev/tap/encore`
- **Linux:** `curl -L https://encore.dev/install.sh | bash`
- **Windows:** `iwr https://encore.dev/install.ps1 | iex`

When you have installed Encore, run:

```bash
encore app create --example=ts/template-engine
```

## Running locally

```bash
encore run
```

**Pages:**

- <http://localhost:4000/person> - Serving a specific view with data
- <http://localhost:4000/> - Serving a template based on the dynamic path
- <http://localhost:4000/about/contact> - Serving a template located in a sub-folder
- <http://localhost:4000/html> - Serving an inline HTML template

> Go to [http://localhost:4000](http://localhost:4000) in the browser to see the example frontend.

Reviewer: ```suggestion
Open [http://localhost:4000](http://localhost:4000) in your browser to see the example frontend.
```

> You can also access Encore's [local developer dashboard](https://encore.dev/docs/observability/dev-dash) on <http://localhost:9400/> to view traces, API documentation, and more.

Reviewer: ```suggestion
You can also access Encore's [local development dashboard](https://encore.dev/docs/observability/dev-dash) on <http://localhost:9400/> to view traces, API documentation, and more.
```
